### PR TITLE
feat: iterator over branch children

### PR DESCRIPTION
**Pull request by rkrasiuk**
## Description

Create iterator over branch node children and add some utility methods.

### CODE REVIEW

File: src/nodes/branch.rs [hunk 1]
```diff
@@ -193,6 +201,50 @@ impl<'a> BranchNodeRef<'a> {
...
 }
 
+/// Iterator over branch node children.
+#[derive(Debug)]
+pub struct BranchChildrenIter<'a> {
+    pos: u8,
+    end: u8,
+    stack_ptr: usize,
+    state_mask: &'a TrieMask,
+    stack: &'a [Vec<u8>],
```
Comment (DaniPopes) [inline src/nodes/branch.rs]: looks like:
- `pos` and `end` can be just `range: Range<u8> = CHILD_INDEX_RANGE` which is also an iterator
- `stack` and `stack_ptr` can be a single `slice::Iter` from `node.stack[node.first_child_index()..].iter()`

using std iterators is cleaner and more efficient as they're the one of the most optimized part of the standard library :saluting_face:

File: src/nodes/branch.rs [hunk 2]
```diff
@@ -193,6 +201,50 @@ impl<'a> BranchNodeRef<'a> {
...
+}
+
+impl<'a> BranchChildrenIter<'a> {
+    /// Create new iterator over branch node children.
+    pub fn new(node: &BranchNodeRef<'a>) -> Self {
```
Comment (DaniPopes) [inline src/nodes/branch.rs]: should probably be private and a method on BranchNodeRef?

Comment (rkrasiuk) [reply src/nodes/branch.rs]: no harm in having it public

File: src/nodes/branch.rs (hunk 1 above)
Comment (rkrasiuk) [reply src/nodes/branch.rs]: collapsed pos and end, stack ptr can't be an iter like that tho

Comment (DaniPopes) [reply src/nodes/branch.rs]: why not?

Comment (rkrasiuk) [reply src/nodes/branch.rs]: had a brain fart, ye it can

File: src/nodes/branch.rs (hunk 2 above)
Comment (DaniPopes) [reply src/nodes/branch.rs]: right, but you would want `node.children()` rather than `BranchChildrenIter::new(node)`

Comment (rkrasiuk) [reply src/nodes/branch.rs]: that's already exposed via `indexed_children`, but alright, point taken, will hide vis

**Review by DaniPopes — APPROVED** [review]

**Pull request merged**